File: Q7TowerOfHanoi.py
def move(n, fr, to, via):
  if n == 1:
    print "Moving pole {0} to {1}".format(fr, to)
  else:
    move(n - 1, fr, via, to)
    move(1, fr, to, via)
    move(n - 1, via, to, fr)
    
move(2, 1, 3, 2)

File: Q7TowerOfHanoi.java
public class Q6TowerOfHanoi {

	public static void move(int n, int from, int to, int via){
		//base
		if(n == 1){
			System.out.println("Moving pole from " + from + " to " + to);
		} else {
			move(n - 1, from, via, to);
			move(1, from, to, via);
			move(n - 1, via, to, from);
		}
	}
	
	public static void main(String[] args) {
		move(2, 1, 3, 2);
	}

}
